vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -382,6 +387,21 @@ func resourceVSphereVirtualMachine() *schema.Resource {
 							Type:     schema.TypeString,
 							Optional: true,
 							ForceNew: true,
+							Default: "vmxnet3",
+							ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
```

Comment:
You can use [`StringInSlice`](https://github.com/hashicorp/terraform/blob/master/helper/validation/validation.go#L74) from the `validation` package in TF core to make this much easier on the eyes.

The syntax would be here:

`ValidateFunc: validation.StringInSlice(virtualMachineNetworkAdapterTypeAllowedValues, false)`